frontend/src/component/common/Limit/Limit.tsx

```diff
@@ -0,0 +1,144 @@
+import { Box, IconButton, styled, Tooltip, Typography } from '@mui/material';
+import LinearProgress from '@mui/material/LinearProgress';
+import { Link } from 'react-router-dom';
+import WarningIcon from '@mui/icons-material/WarningAmber';
+import ErrorIcon from '@mui/icons-material/Cancel';
+import CloseIcon from '@mui/icons-material/Close';
+import type { FC } from 'react';
+import { ConditionallyRender } from '../ConditionallyRender/ConditionallyRender';
+
+const StyledBox = styled(Box)(({ theme }) => ({
+    display: 'flex',
+    flexDirection: 'column',
+    border: `2px solid ${theme.palette.background.application}`,
+    borderRadius: `${theme.shape.borderRadiusMedium}px`,
+}));
+
+const BorderLinearProgress = styled(LinearProgress)(({ theme }) => ({
+    height: theme.spacing(1.5),
+    borderRadius: theme.shape.borderRadiusMedium,
+}));
+
+const StyledWarningIcon = styled(WarningIcon)(({ theme }) => ({
+    color: theme.palette.warning.main,
+}));
+
+const StyledErrorIcon = styled(ErrorIcon)(({ theme }) => ({
+    color: theme.palette.warning.main,
+}));
+
+const Header = styled(Box)(({ theme }) => ({
+    display: 'flex',
+    gap: theme.spacing(1),
+    alignItems: 'center',
+    fontWeight: 'bold',
+    borderBottom: `2px solid ${theme.palette.background.application}`,
+    padding: theme.spacing(3, 4),
+    fontSize: theme.fontSizes.mainHeader,
```

Comment:
Isn't `fontSizes` deprecated? Can we use `theme.typography` instead? something like:

```suggestion
    fontSize: theme.typography.h2.fontSize,
```

Reply:
didn't know about this one

Reply:
Oh, yeah! I found it a while back when I got sick of seeing that `fontSizes` was deprecated 🤷🏼  Not sure if that's the right size for you, though! But it's probably one of them.